packages/hoppscotch-sh-admin/src/modules/admin.ts

```diff
@@ -1,14 +1,19 @@
 import { auth } from '~/helpers/auth';
 import { HoppModule } from '.';
 
-const isAdmin = () => {
-  const user = auth.getCurrentUser();
-  return user ? user.isAdmin : false;
-};
+const isSetupRoute = (to: unknown) => to === 'setup';
+
+const isGuestRoute = (to: unknown) => ['index', 'enter'].includes(to as string);
 
-const GUEST_ROUTES = ['index', 'enter'];
+const getAdminStatus = async () => {
+  const user = await auth.checkCurrentUser();
+  return !!user?.isAdmin;
+};
 
```

Comment:
Attempting to visit a route from any of the nav items with an invalid access token redirects to the login page at first and returns to the dashboard on a page reload. In the existing implementation, `auth.getCurrentUser()` would hold on to the user stream value in such a scenario, but a network call via `auth.checkCurrentUser()` would fail as in the current implementation. We'd want to keep the current implementation (network calls as a single source of truth) by handling the above situation.

Reply:
Resolved via https://github.com/hoppscotch/hoppscotch/pull/3845/commits/30362c1c61babe77f06fece8c329293b84af4654.